src/util-plugin.c

```diff
@@ -211,4 +212,32 @@ SCCapturePlugin *SCPluginFindCaptureByName(const char *name)
     }
     return plugin;
 }
+
+#ifdef ALPROTO_DYNAMIC_NB
+static SCAppLayerPlugin *app_layer_plugins[ALPROTO_DYNAMIC_NB];
+static size_t app_layer_plugins_nb = 0;
+#endif
+
+int SCPluginRegisterAppLayer(SCAppLayerPlugin *plugin)
```

Comment:
I'd prefer to not think about registering an app-layer as a plugin. Could app-layers not be registered the same being included or as plugins?  For plugin support we just expose the API that Suricata would already be using.

Reply:
I do not understand your sentences here... Could you rephrase ?

(Plus I need to push a plugin example)

Reply:
Why couldn't a non-plugin AppLayer not register itself through this API? In which case its no longer a plugin API, its just the API for registering plugin.  In short, putting Plugin in here seems a little artificial.

It was a mistake in the EVE file type API which I've already started to undo.

Reply:
Ok, so this is just about renaming `SCPluginRegisterAppLayer` to `SCRegisterDynamicAppLayer`, do I get it ?

Would we want to get rid of all the static ones ?

Reply:
> Would we want to get rid of all the static ones ?

Personally I'd think so unless there is a performance hit?

I like to think of plugin support as just exposing the APIs we already use internally.  Then there is nothing really different from a built-in or plugin, and we're constantly dog-fooding the APIs.

Reply:
And so, should this be moved out of `util-plugin.c` to `app-layer-register.c` ?